### DISCUSSION TITLE
Is it possible to get a piece location by its name and color?

### DISCUSSION THREAD

**Discussion by HarshilPatel007** [niklasf/python-chess#1069] [Q&A] [answered]
Hello @niklasf and community,

I want to know if there is a built-in function to get the location in (x, y) format of a given piece name and color.
Example: get_piece_location("w","R")
Output like this, [(7,7), (0,7)]

Use-case,
I'm developing a chess gui (https://github.com/HarshilPatel007/YACS)
Where I just delete a piece image at the source square (piece moved from) and add piece image at the destination square (piece moved to) after the move has been played instead of deleting all the pieces and re-drawing them.

I've tackled the en-passent and both side castling cases for the normal chess variant but the real problem arises with the chess960 variant. How, am I going to get the position of the rooks and King?
Because, to specifically handle the casting case, I want to know the location of rook and king before and after casting in order to place them on the right square.

**Comment by niklasf** [comment, marked as answer]
```python
import chess

board = chess.Board()

for sq in board.pieces(chess.ROOK, chess.WHITE):
    print(chess.square_name(sq), chess.square_file(sq), 7 - chess.square_rank(sq))
```

Would print:

```
a1 0 7
h1 7 7
```

`7 - chess.square_rank(sq)` because ranks increase as you go up the board, whereas the coordinates you mention in your question are flipped.

**Reply by HarshilPatel007** [reply]
Hi @niklasf ,

Thanks. I've got things working. However, is there a way to get the starting position FEN of chess960?
However, I know the ways of getting the current or starting position of the normal variant. Sorry for being dumb.

**Reply by niklasf** [reply]
To understand what you're asking... what kind of obiect/file do you have, that you want to extract the starting FEN from?

**Reply by HarshilPatel007** [reply]
i just want to get the fen of first starting position of a chess960 chessboard. (it should be static string just like `chess.STARTING_BOARD_FEN`)

**Reply by niklasf** [reply]
Mhh ... but there are many possible starting position, so it can't be a constant like that. Assuming you already have a `board` of the specific one, you can do `board.fen()` (`board.root().fen()` after some moves were already played).

If not, then `board = chess.Board.from_chess960_pos(n)` will get the nth one, or `board = chess.Board.from_chess960_pos(random.randint(0, 959))` a random one.

**Reply by HarshilPatel007** [reply]
@niklasf , I think you got it wrong.
I mean, the board is already set as chess960 and up and running. I wanted to get the starting fen of that chessboard using `board.fen()` but, only once.
However, I got it. I just called it once after the board was initiated and stored that in a class. So, the problem is solved.
Earlier, I was confused, because `board.fen()` returned a new updated value after the move. (It should 🙂 and it does! 😁)
